Introduce build-component-images as step to post-monorepo joy

jchesterpivotal opened the pull request:
Pretty narrowly focused on just the minimum here. I expect I'll be able to follow with PRs to tear out old jobs, tasks and resources.

jchesterpivotal committed:
Update sidecar/build task to use monorepo paths

- Simplify version selection
- Remove vestigial reference to docker-context

[Github #52]

jchesterpivotal committed:
Introduce build-component-images job

Previously, each server component of riff (http-gateway,
topic-controller, function-controller, function-sidecar) is built in
separate jobs, with separate build numbers. This largely comes about
because of the previous multirepo world. Moving to a monorepo
(projectriff/riff#296) means that we can reduce our bookkeeping and
reduce our collection of tasks.

Because it needs Kafa running, function-sidecar remains on an updated
sidecar/build task. The others were switched to a common task
(riff-components/test), introduced in previous commits.

The job introduced by this commit does not successfully build against
the riff master branch, it should do so once projectriff/riff#306 has
been merged.

This commit does not replace the existing jobs, does not trigger on
git commits and does not remove unused tasks.

- Add build-component-images job
- Add riff-version resource
- Add github-commit-status-riff resource
- Remove spurious go get mockery from riff-components/test/run

[Github #52]

ericbottard commented on pipelines/riff.yml:
```diff
@@ -460,6 +476,89 @@ jobs: ##########################################################################
+        description: Building, testing and creating docker images for riff server components
+
+  - aggregate: # build and test
+    - task: test-topic-controller
```
Ah, scratch that. I see docker push is after

ericbottard commented on pipelines/riff.yml:
```diff
@@ -460,6 +476,89 @@ jobs: ##########################################################################
+
+  - aggregate: # build and test
+    - task: test-topic-controller
+      file: git-pfs-ci/tasks/riff-components/test/task.yml
```
I thought we agreed on not trying to factorize behavior across components.
I believe this PR will work, but then we should at least clean unused files

jchesterpivotal replied:
Only sidecar needs the kafka container, all the others work with the `test` task. I did notice that while I updated that task (`test-function-sidecar`) in the `riff` pipeline, it still have a `params` block hanging off from when I tried to do it using the factorized `test` task.

I had in mind to push the cleanup into a following commit. Not that I don't want to do it, but I'd like to get to the working case first.

jchesterpivotal committed:
Remove vestigial params from test-function-sidecar

- Move to bottom for visual distinction from jobs using the
  riff-components/test task

[Github #52]

ericbottard commented:
LGTM then.
I am about to merge this, is this ok ?

Also, I noticed you used the projectriff/ci repo to issue your PR. We usually keep those "reference" repos clean and issue PRs from forks

jchesterpivotal commented:
Yeah, I misstepped on the workflow. Happy for the merge to go ahead.
